autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnResponseUtils.java

```diff
@@ -23,33 +21,15 @@
 
     public List<String> getYarnRecommendedDecommissionHostsForHostGroup(String clusterCrn, YarnScalingServiceV1Response yarnResponse,
             Map<String, String> hostFqdnsToInstanceId, int maxAllowedDownScale, Optional<Integer> mandatoryDownScaleCount) {
-        Set<String> consideredNodeIds = new HashSet<>();
-        Integer allowedDownscale = Math.max(maxAllowedDownScale, mandatoryDownScaleCount.orElse(0));
+        Integer allowedDownscale = mandatoryDownScaleCount.orElse(maxAllowedDownScale);
```

Comment:
Looking at this independently, it would seem that if mandatoryDownscale is not present, we default to maxAllowedDownscale.
That isn't actually what happens with loadBased downscaling. If there's no mandatory downscale, the request to YARN would not have maxAllowedDownscale candidates. So reading this function independently can be quite confusing.
I think it's worthwhile noting that "yarnResponse" passed to this function may or may not contain maxAllowedDownscale candidates.

Reply:
maxAllowedDownscale is always passed, even in loadBased downscaling without mandatory downscale that is within limit, maxAllowedDownScale is passed based on policy limit.

Reply:
Added comment regarding yarnResponse might not have maxAllowedDownscale candidates